lib/internal/modules/cjs/loader.js

```diff
@@ -359,10 +360,22 @@ function setModuleParent(value) {
   moduleParentCache.set(this, value);
 }
 
-let debug = require('internal/util/debuglog').debuglog('module', (fn) => {
+const { debuglog, isDebugEnabled } = require('internal/util/debuglog');
+
+let debug = debuglog('module', (fn) => {
   debug = fn;
 });
 
+let hrtimeBigIntTimingFn = () => {
+  hrtimeBigIntTimingFn = isDebugEnabled('timing_module_cjs') ? process.hrtime.bigint : () => 0;
```

Comment:
I think it would make more sense to implement this in debuglog. Something like:

```js
const { debuglogWithTimer } = require('internal/util/debuglog');
const timer = debuglogWithTimer('timing_module_cjs');

// Say in `Module._load`, 'mark' should be request i.e. the module specifier/filename
timer.startTimer('mark');
// do some stuff
timer.logTimer('mark');
// do some stuff
timer.logTimer('mark');
// do some stuff
timer.logTimer('mark');
```

Then it'll be easy to reuse this in other places. And the `debuglogWithTimer` can emit traces via `internalBinding('trace_events').trace`.

Reply:
I liked the idea but I didn't like the API, if we do it this way, we need to keep the start time in memory to be able to have the message `took 2ms`.

Or, we can just log the `hrtime` in every log and remove the necessity of `startTimer`:

```js
const mark = `[${ parent?.id }] [${ request }]`;

logTimer(mark, 'started');
// load the module
logTimer(mark, 'finished');
```

```
{debug_module} {pid}: HRTIME [{hrtime}] {mark} {message} 
```

In this way, what you did will be transformed as:

```
TIMING_MODULE_CJS 114493: HRTIME [1000] [/home/user/project/file.js] [path] started
TIMING_MODULE_CJS 114493: HRTIME [5000] [/home/user/project/file.js] [path] finished
```

Reply:
I tried to create an implementation using `trace` and I don't think this is a good idea anymore because to be able to emit traces, we need to run the application like:

```
NODE_DEBUG=timing_module_cjs node --trace-categories node.timing_module index.js
```

And we can't enable the `trace` without enabling the `NODE_DEBUG`.

If we want to do that, we need to change how the `NODE_DEBUG` is enabled to make it work (and not set `noop`) when both `NODE_DEBUG` or `trace_events` is enabled.

Do you think that is worth implementing?